### PR TITLE
Test send specific receipts (Part 2)

### PR DESCRIPTION
**Pull request by gwen917**
## Why this should be merged
Implements the following test cases from https://github.com/ava-labs/teleporter/issues/94
- Relayer manually retries sending of specific receipts.

## How this works
See above

## How this was tested
New E2E test cases

## How is this documented

### CODE REVIEW

File: tests/send_specific_receipts.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	"github.com/ava-labs/teleporter/tests/utils"
+	"github.com/ethereum/go-ethereum/common"
+	. "github.com/onsi/gomega"
+)
+
+func SendSpecificReceiptsGinkgo() {
+	sendSpecificReceipts(&network.LocalNetwork{})
+}
+
+func sendSpecificReceipts(network network.Network) {
```
Comment (cam-schultz) [inline tests/send_specific_receipts.go:19]: Let's export this function

File: tests/send_specific_receipts.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	// Check delivered
+	delivered, err = subnetATeleporterMessenger.MessageReceived(&bind.CallOpts{}, subnetBInfo.BlockchainID, messageID)
+	Expect(err).Should(BeNil())
+	Expect(delivered).Should(BeTrue())
+
+	// Check the reward amount. The reward amount should be 10
+	amount, err := subnetATeleporterMessenger.CheckRelayerRewardAmount(&bind.CallOpts{}, fundedAddress, mockTokenAddress)
+	Expect(err).Should(BeNil())
+	Expect(amount).Should(Equal(big.NewInt(10)))
+
```
Comment (cam-schultz) [inline tests/send_specific_receipts.go:98]: Can you add an additional step of sending an arbitrary message from Subnet B -> Subnet A? This message will also include the receipts that were explicitly sent in the above steps. We should check that the relayer reward amount is unchanged, even after the receipt is delivered again.

Comment (gwen917) [reply tests/send_specific_receipts.go:98]: sure. Then, will it cover `Test receiving a receipt multiple times has no effect` case?

File: tests/send_specific_receipts.go (hunk 1 above)
Comment (gwen917) [reply tests/send_specific_receipts.go]: fixed

File: tests/send_specific_receipts.go (hunk 2 above)
Comment (cam-schultz) [reply tests/send_specific_receipts.go:98]: Yup it will! I pointed it out for completeness here, but that will also cover that test case.

**Review by cam-schultz — APPROVED** [review]

**Pull request merged**